cocotb/share/lib/gpi/gpi_priv.h

```diff
@@ -125,6 +131,9 @@ class GpiObjHdl : public GpiHdl {
     virtual const char* get_type_str(void);
     gpi_objtype_t get_type(void) { return m_type; };
     bool get_const(void) { return m_const; };
+    bool get_is_port(void) { return m_is_port; };
+    virtual const char* get_port_direction_str(void);
```

Comment:
Why is this virtual?

Reply:
Thanks for the review, I now made this a helper function.
Your enum suggestion (I like enums! :) ) should probably be a new issue so that adding a dependency can be discussed.